capreolus/index/faiss.py

```diff
@@ -0,0 +1,33 @@
+from capreolus import ConfigOption, constants, get_logger
+
+from . import Index
+
+
+logger = get_logger(__name__)
+
+
+@Index.register
+class FAISSIndex(Index):
+    module_name = "faiss"
+    
+    dependencies = [Dependency(key="encoder", module="encoder"), Dependency(key="index", module="index")] + Index.dependencies
+
+    def _create_index(self):
+        collection_path = self.collection.get_path_and_types()
+        # 1. Create the anserini index from collection
+        # 2. Pass each document in the index through the encoder to get the vector rep
+        #   a. There's some scope for batching - it would be extremely slow otherwise
+        #   b. We'll call "self.encoder.encode(doc)" here in a for-loop
+        # 3. Map the string doc-id to an int64 id (FAISS can handle only integer ids). This would create a one-to-one map between vectors in the FAISS index and the raw docs in anserini
```

Comment:
Keeping this map is annoying. I read somewhere in the repo that there's a sqlite DB integration coming - if that's not done already, maybe I can do that? :)

Reply:
I think @crystina-z has an example of sqlite with an extractor, so that would work as one option.

It looks like you're already going to build an Anserini index though, so what if we just use the mapping provided by that Lucene index? Anserini translates between the "real" string docids and a int id used internally. I think pyserini exposes this somewhere. That would avoid the needing to build up the mapping incrementally.

Reply:
Ah that's great - I'll just use the lucene ID then. Thanks 👍